File: 0415-add-strings/0415-add-strings.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var addStrings = function(num1, num2) {

    if(num1.length==1 && num2.length==1){
        return (parseInt(num1) + parseInt(num2)).toString();
    }

    let s1 = num1.split("").reverse().join("");
    let s2= num2.split("").reverse().join("");
    
    let i=0;
    let j=0;

    let ans="";
    let carry=0;
    while(i<s1.length && j<s2.length){
        
      let temp = parseInt(s1[i]) + parseInt(s2[j]) + carry;
      if(temp>=10){
          temp=temp%10;
          ans+=temp;
          carry=1;
          i++;
          j++;

      }

      else{
          ans+=temp;
          i++;
          j++;
          carry=0;
      }

     
    }

    while(i<s1.length){
        let temp=0;
        if(carry){
            temp+= parseInt(s1[i]) + carry;
            if(temp>=10){
                temp = temp%10;
                ans+=temp;
                carry=1;
                i++;
            }

            else{
                ans+=temp;
                i++;
                carry=0;
            }
            
        }
        else{
           ans+=s1[i];
           i++;
        }
       
    }

    while(j<s2.length){
       let temp=0;
        if(carry){
            temp+= parseInt(s2[j]) + carry;
            if(temp>=10){
                temp = temp%10;
                ans+=temp;
                carry=1;
                j++;
            }

            else{
                  ans+=temp;
                  j++;
                  carry=0;
            }
            
        }
        else{
           ans+=s2[j];
           j++;
        }
    }

    if(carry){
           ans+=carry;
    }

   let sum =  ans.split("").reverse().join("");

     return sum;

};;

File: 0415-add-strings/0415-add-strings.cpp
class Solution {
public:

string add(string num1 , string num2){
    int carry =0;
    int i = num1.length()-1;
    int j=num2.length()-1;
    string ans="";
    
    while(j<num2.length()){
        int sum = (num1[i] - '0') + (num2[j]-'0') + carry;

       
        carry = sum/10;

        char c = '0'+sum%10; // converting int sum%10 to char 'sum%10'

        ans+=c;

        i--;
        j--;



    }


    while(i<num1.length()){
           int sum = (num1[i] - '0') + carry;
            carry = sum/10;

            char c = '0'+sum%10; // converting int sum%10 to char 'sum%10'

             ans+=c;

              i--;
    
    }



    if(carry){
        ans+='1';
    }


    reverse(ans.begin() ,ans.end());


    return ans;
}















    string addStrings(string num1, string num2) {
    
     if(num1.length() >= num2.length()){
        return add(num1 , num2);
     }

     else{
        return add(num2 , num1);
     }

     
    }
};